core/modules/nat.h

```diff
@@ -0,0 +1,98 @@
+#ifndef BESS_MODULES_NAT_H_
+#define BESS_MODULES_NAT_H_
+
+#include <arpa/inet.h>
+
+#include <map>
+#include <utility>
+#include <vector>
+
+#include "../module.h"
+#include "../module_msg.pb.h"
+#include "../utils/ip.h"
+#include "../utils/random.h"
+
+using bess::utils::IPAddress;
+using bess::utils::CIDRNetwork;
+
+typedef std::pair<CIDRNetwork, CIDRNetwork> NATRule;
+
+template <class T>
+inline void hash_combine(std::size_t &seed, const T &v) {
+  std::hash<T> hasher;
+  seed ^= hasher(v) + 0x9e3779b9 + (seed << 6) + (seed >> 2);
+}
+
+class NAT final : public Module {
+ public:
+  struct Flow {
```

Comment:
Is there a reason struct Flow is declared inside the NAT class?  Also, Since it has a non-trivial method, perhaps it should be declared as a Class and add a comment?